hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/processors/StreamToStreamJoinP.java

```diff
@@ -325,22 +331,44 @@ public boolean saveToSnapshot() {
                 }
             }
 
-            Stream<Entry<?, ?>> leftBufferStream = buffer[0].content()
+            leftBufferStream = buffer[0].content()
                     .stream()
                     .map(row -> entry(
-                            ObjectArrayKey.project(row, joinInfo.leftEquiJoinIndices()),
+                            joinInfo.isEquiJoin() ? project(row, joinInfo.leftEquiJoinIndices()) : row.hashCode(),
```

Comment:
In this specific keys I'd keep the `ObjectArrayKey.` prefix before `project`. Without it, the `project` name is too generic and the reader cannot know that it's a static factory method for an `ObjectArrayKey` instance.